Build/Tests: enable testing against PHP 8.1 (Trac 53891)

jrfnl opened the pull request:
PHP 8.1 is expected to be released end of November. Enabling the tests to run in CI on PHP 8.1 will allow us to get ready in time.

These builds are, for now, still allowed to fail.



Trac ticket: https://core.trac.wordpress.org/ticket/53891

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

jrfnl commented:
Once the patch is committed, all builds will show as "failed" until all PHP 8.1 issues have been fixed, which has the risk of people not noticing new failures on PHP < 8.1.

As an interim measure, while working through the PHP 8.1 issues, we could add separate conditional steps to run the tests on PHP 8.1 and add `continue-on-error` to those. That would allow the test builds to show as "successful" if all non-PHP 8.1 test runs pass.

Opinions ? /cc @desrosj

SergeyBiryukov commented:
> As an interim measure, while working through the PHP 8.1 issues, we could add separate conditional steps to run the tests on PHP 8.1 and add `continue-on-error` to those. That would allow the test builds to show as "successful" if all non-PHP 8.1 test runs pass.

That makes sense to me 👍

SergeyBiryukov commented:
> As an interim measure, while working through the PHP 8.1 issues, we could add separate conditional steps to run the tests on PHP 8.1 and add `continue-on-error` to those. That would allow the test builds to show as "successful" if all non-PHP 8.1 test runs pass.

One other consideration I've just learned from @aristath is that if WordPress core tests fail, Gutenberg contributors cannot merge any PRs, so definitely let's do that :)

jrfnl commented:
* Rebased after the patches for the blockers had been merged.
* Updated to apply my earlier suggestion.
    You can see that the builds against PHP 8.1 now show as 🟢 , even though when you open them up, you will see that all test runs have failures.
    The failures are also listed on the CI "Summary" overview of the "PHPUnit Tests" workflow.
    See: https://github.com/WordPress/wordpress-develop/actions/runs/1113845520

![image](https://user-images.githubusercontent.com/663378/128755722-498bf641-3098-4ada-8bc9-0d4502e9b089.png)

jrfnl commented:
Closing as committed in [51588](https://core.trac.wordpress.org/changeset/51588)
